providers/netty4/src/main/java/org/asynchttpclient/netty/request/body/NettyReactiveStreamBody.java

```diff
@@ -0,0 +1,140 @@
+    private static final String NAME_IN_CHANNEL_PIPELINE = "request-body-streamer";
+
+    private final Publisher<ByteBuffer> publisher;
+
+    public NettyReactiveStreamBody(Publisher<ByteBuffer> publisher) {
+        this.publisher = publisher;
+    }
+
+    @Override
+    public long getContentLength() {
+        return -1L;
+    }
+
+    @Override
+    public String getContentType() {
+        return null;
+    }
+
+    @Override
+    public void write(Channel channel, NettyResponseFuture<?> future) throws IOException {
+        if (future.isStreamWasAlreadyConsumed()) {
+            LOGGER.warn("Stream has already been consumed and cannot be reset");
+        } else {
+            future.setStreamWasAlreadyConsumed(true);
+            NettySubscriber subscriber = new NettySubscriber(channel);
+            channel.pipeline().addLast(NAME_IN_CHANNEL_PIPELINE, subscriber);
+            publisher.subscribe(new SubscriberAdapter(subscriber));
+        }
+    }
+
+    private static class SubscriberAdapter implements Subscriber<ByteBuffer> {
+        private volatile Subscriber<HttpContent> subscriber;
+        
+        public SubscriberAdapter(Subscriber<HttpContent> subscriber) {
+            this.subscriber = subscriber;
+        }
+        @Override
+        public void onSubscribe(Subscription s) {
+           subscriber.onSubscribe(s);
+        }
+        @Override
+        public void onNext(ByteBuffer t) {
+            ByteBuf buffer = Unpooled.wrappedBuffer(t.array());
+            HttpContent content = new DefaultHttpContent(buffer);
+            subscriber.onNext(content);
+        }
+        @Override
+        public void onError(Throwable t) {
+            subscriber.onError(t);
+        }
+        @Override
+        public void onComplete() {
+            subscriber.onComplete();
+        }        
+    }
+    
+    private static class NettySubscriber extends HandlerSubscriber<HttpContent> {
+        private static final Logger LOGGER = LoggerFactory.getLogger(NettySubscriber.class);
+
+        private final Channel channel;
+
+        public NettySubscriber(Channel channel) {
+            super(channel.eventLoop());
+            this.channel = channel;
+        }
+
+        @Override
+        public void onComplete() {
+            EventExecutor executor = channel.eventLoop();
+            executor.execute(new Runnable() {
+                @Override
+                public void run() {
+                    channel.writeAndFlush(LastHttpContent.EMPTY_LAST_CONTENT).addListener(new ChannelFutureListener() {
+                        @Override
+                        public void operationComplete(ChannelFuture future) throws Exception {
+                            removeFromPipeline();  
+                        }
+                    });
+                }
+            });
+        }
+
+        @Override
+        public void onError(Throwable error) {
```

Comment:
Same with `onComplete`, `HandlerSubscriber` provides an `error` method for the purposes of overriding.